src/cdrizzlemap.c

```diff
@@ -511,162 +475,109 @@ orient_ccw(struct polygon *p) {
-            }
+    wv_ = (struct vertex *)(wnd->v + (wnd->npv - 1));
+    wv = (struct vertex *)wnd->v;
 
-            if (pv_in_hpdq) {
-                inside = 1;
-            } else if (qv_in_hpdp) {
-                inside = -1;
-            }
-        }
+    for (k = 0; k < wnd->npv; k++) {
+        dw.x = wv->x - wv_->x;
+        dw.y = wv->y - wv_->y;
 
-        // advance:
-        if (d < 1.0e-12 && !pv_in_hpdq && !qv_in_hpdp) {
-            if (inside == 1) {
-                iq += 1;
-                qv_ = qv;
-                qv = q->v + mod(iq, q->npv);
-            } else {
-                ip += 1;
-                pv_ = pv;
-                pv = p->v + mod(ip, p->npv);
-            }
+        // use output from previous iteration as input for the current
+        tpp = ppin;
+        ppin = ppout;
+        ppout = tpp;
+        ppout->npv = 0;
 
-        } else if (signed_area >= 0.0) {
-            if (qv_in_hpdp) {
-                if (inside == 1) {
-                    if (append_vertex(pq, *pv)) break;
-                }
-                ip += 1;
-                pv_ = pv;
-                pv = p->v + mod(ip, p->npv);
-            } else {
-                if (inside == -1) {
-                    if (append_vertex(pq, *qv)) break;
-                }
-                iq += 1;
-                qv_ = qv;
-                qv = q->v + mod(iq, q->npv);
-            }
+        pv_ = (struct vertex *)(ppin->v + (ppin->npv - 1));
+        pv = (struct vertex *)ppin->v;
 
-        } else {
-            if (pv_in_hpdq) {
-                if (inside == -1) {
-                    if (append_vertex(pq, *qv)) break;
-                }
-                iq += 1;
-                qv_ = qv;
-                qv = q->v + mod(iq, q->npv);
-            } else {
-                if (inside == 1) {
-                    if (append_vertex(pq, *pv)) break;
+        for (j = 0; j < ppin->npv; j++) {
+            dp.x = pv->x - pv_->x;
+            dp.y = pv->y - pv_->y;
+
+            v1_inside = is_point_strictly_in_hp(*wv_, *wv, *pv_);
+            v2_inside = is_point_strictly_in_hp(*wv_, *wv, *pv);
+
+            if (v2_inside != v1_inside) {
+                // compute intersection point:
+                // https://en.wikipedia.org/wiki/Line–line_intersection
+                t = (pv_->y - wv_->y) * dw.x - (pv_->x - wv_->x) * dw.y;
+                // u = (pv_->y - wv_->y) * dp.x - (pv_->x - wv_->x) * dp.y;
+                d = area(dp, dw);  // d != 0 because (v2_inside != v1_inside)
+
+                t = t / d;
+                // u = u / d;
+
+                vi.x = pv_->x + (pv->x - pv_->x) * t;
+                vi.y = pv_->y + (pv->y - pv_->y) * t;
+                // vi.x = 0.5 * (pv_->x + wv_->x + (pv->x - pv_->x) * t +
+                //               (wv->x - wv_->x) * u);
+                // vi.y = 0.5 * (pv_->y + wv_->y + (pv->y - pv_->y) * t +
+                //               (wv->y - wv_->y) * u);
```

Comment:
Are these lines necessary? And a few lines above?

Reply:
These are not necessary but I am torn on using them. Right now we compute the intersection point along just one line. The point computed along the second line _should_ give the same intersection point. So, to reduce computational errors, it seems it would be better to average the two estimates. However, for our purposes the accuracy from one line, I think, should be enough. So, for now I would like to keep this code commented and maybe I should add a comment to explain what it is doing (or what it would do if uncommented) just in case we encounter a situation that might need it later.